library/core/src/future/future.rs

```diff
@@ -28,10 +28,13 @@ use crate::task::{Context, Poll};
 #[must_use = "futures do nothing unless you `.await` or poll them"]
 #[stable(feature = "futures_api", since = "1.36.0")]
 #[lang = "future_trait"]
-#[diagnostic::on_unimplemented(
-    label = "`{Self}` is not a future",
-    message = "`{Self}` is not a future",
-    note = "{Self} must be a future or must implement `IntoFuture` to be awaited"
+#[cfg_attr(
```

Comment:
Why did you `cfg_attr` it here? It doesn't appear to have changed. You should `cfg_attr` the `feature()` declaration instead.

Reply:
I just did not occur to me that that would also work :see_no_evil: 
I've pushed a new version that reverts the changes and uses that approach instead.